library/common/http/client.cc

```diff
@@ -80,6 +80,25 @@ void Client::DirectStreamCallbacks::encodeData(Buffer::Instance& data, bool end_
     closeStream();
   }
 
+  // The response_data_ is systematically assigned here because resumeData can
+  // incur an asynchronous callback to sendDataToBridge. It might look like
+  // safe to only set the response_data_ after invoking sendDataToBridge in this
```

Comment:
Thanks for explaining this one to me.  I think you can cut this down to just the top two lines as that's sufficient for me at least :-)